src/application/user_settings.rs

```diff
@@ -0,0 +1,94 @@
+use serde::{Serialize, Deserialize};
+use serde_json;
+use std::env;
+use std::path::PathBuf;
+use std::collections::HashMap;
+use crate::application::gactions::Theme;
+
+#[derive(Serialize, Deserialize, Hash, PartialEq, Eq)]
+pub enum UserSettingsKey {
+    ThemeKey,
+}
+
+#[derive(Serialize, Deserialize)]
+pub enum UserSettingsValue {
+    ThemeValue(Theme)
+}
+
+pub struct UserSettings {
+    user_settings: HashMap<UserSettingsKey, UserSettingsValue>,
+    config_name: String,
+}
+
+impl Default for UserSettings {
+    fn default() -> Self {
+        let mut instance = UserSettings::new();
+
+        if let Err(_load_config_err) = instance.load_config() {
+            match instance.create_config() {
+                Ok(_) => (),
+                Err(_create_config_err) => {
+                    println!("Failed to create config, saving will not be enabled");
+                }
+            }
+        }
+
+        instance
+    }
+}
+
+impl UserSettings {
+    fn new() -> Self {
+        Self {
+            user_settings: HashMap::new(),
+            config_name: "config.json".to_string(),
```

Comment:
Sadly, we cannot expect that the current working directory is always writable, we have to use the platform's config directory.

- **Linux** and **MacOS**: if defined, use `$XDG_CONFIG_HOME/logicrs/config.json` else `$HOME/.config/logicrs/config.json`
- **Windows**: use `%USERPROFILE%\AppData\Local\logicrs\config.json`

(environment variables need to be expanded)

Reply:
Got it, thanks! Will make that tweak. Just as an FYI this PR is still draft form so not all functionality is there (but thank you for taking the time to have a look already!)

Reply:
Yea, no pressure, just wanted to inform you :)